src/Compilers/CSharp/Portable/Symbols/TypeWithAnnotations.cs

```diff
@@ -265,7 +265,7 @@ internal TypeWithAnnotations MergeEquivalentTypes(TypeWithAnnotations other, Var
         /// <summary>
         /// The list of custom modifiers, if any, associated with the <see cref="Type"/>.
         /// </summary>
-        public ImmutableArray<CustomModifier> CustomModifiers => _extensions.CustomModifiers;
+        public ImmutableArray<CustomModifier> CustomModifiers => _extensions?.CustomModifiers ?? default;
```

Comment:
>_extensions?.CustomModifiers ?? default; [](start = 65, length = 40)

Has something changed? Why do we need to add this logic in this PR?

Reply:
CI flagged an issue (null ref crash)

Reply:
in this assertion in SignatureOnlyMethodSymbol:
`Debug.Assert(isInitOnly == CustomModifierUtils.HasIsExternalInitModifier(returnType.CustomModifiers));`

Reply:
@AlekseyTs Two more tests failed in CI, desktop only. I'm pushing fixes now. Thanks

Reply:
We hit this problem in  `PerformCrefOverloadResolution` which does pass a `default` type.
The reason I'm fixing the issue in `TypeWithAnnotations` is that it mostly already handled when `_extensions` is null. Do you think it would be better to return an empty array instead?

Reply:
> We hit this problem in  PerformCrefOverloadResolution which does pass a default type.

It looks like the comparer that is used there is supposed to ignore return type and also is supposed to ignore custom modifiers. So. I think we need to understand why exactly do we get here for the default instance, what are we trying to do and why.

> The reason I'm fixing the issue in TypeWithAnnotations is that it mostly already handled when _extensions is null. Do you think it would be better to return an empty array instead?

I am not sure if we even should touch this property. Default instances of this structure are not safe by design, I see other properties that will throw or return something that consumers would not expect normally. We simply shouldn't be operating on them.

---
In reply to: [418013638](https://github.com/dotnet/roslyn/pull/43275#discussion_r418013638) [](ancestors = 418013638)